buildSrc/src/main/java/net/neoforged/neodev/GenerateFinalizeSpawnTargets.java

```diff
@@ -0,0 +1,80 @@
+package net.neoforged.neodev;
+
+import com.google.gson.GsonBuilder;
+import net.neoforged.neodev.utils.AsmUtils;
+import net.neoforged.neodev.utils.FileUtils;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * This task is used to generate targets for the finalizeSpawn MethodRedirector coremod.
+ */
+public abstract class GenerateFinalizeSpawnTargets extends DefaultTask {
+    @InputFile
+    public abstract RegularFileProperty getInput();
+
+    @OutputFile
+    public abstract RegularFileProperty getOutput();
+
+    @TaskAction
+    public void exec() throws IOException {
+        var visitor = new Visitor();
+        AsmUtils.visitAllClasses(
+                getInput().getAsFile().get(),
+                visitor,
+                ClassReader.SKIP_DEBUG | ClassReader.SKIP_FRAMES);
+
+        var classList = List.copyOf(visitor.matchedClasses);
+
+        FileUtils.writeStringSafe(
+                getOutput().getAsFile().get().toPath(),
+                new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create().toJson(classList),
+                StandardCharsets.UTF_8);
+    }
+}
+
+class Visitor extends ClassVisitor {
```

Comment:
This should be an inner class, otherwise it is going to be in `net.neoforged.neodev.Visitor`